lambdas/services/create_document_reference_service.py

```diff
@@ -51,10 +55,24 @@ def create_document_reference_request(
         lg_documents_dict_format: list = []
         url_responses = {}
 
+        upload_request_documents = self.parse_documents_list(documents_list)
+
+        has_lg_document = any(
+            document.docType == SupportedDocumentTypes.LG
+            for document in upload_request_documents
+        )
+
```

Comment:
Would it make sense to have those in side the loop?
```suggestion
        for document in documents_list:
          parsed_document = self.parse_documents_list(document)
          upload_request_documents.append(parsed_document)
          has_lg_document = (document.docType == SupportedDocumentTypes.LG) or has_lg_document 


```